### PR TITLE
In IBM Cloud if COS cred secret is missing use pv-pool for default backing store

### PR DESCRIPTION
**Pull request by ambiknai**
Signed-off-by: Ambika Nair <ambiknai@in.ibm.com>
This PR enables to use pv-pool for default backing store in IBM Cloud if COS cred secret is missing

### CODE REVIEW

**Comment by ambiknai** [comment]
```
time="2021-02-09T11:30:51Z" level=info msg="✈️  RPC: redirector.register_to_cluster() Request: <nil>"
time="2021-02-09T11:30:51Z" level=info msg="✅ RPC: redirector.register_to_cluster() Response OK: took 0.3ms"
time="2021-02-09T11:30:51Z" level=info msg="❌ Not Found: BackingStore \"noobaa-default-backing-store\"\n"
time="2021-02-09T11:30:51Z" level=info msg="IBM objectstore credentials \"ibm-cloud-cos-creds\" created. Creating default backing store on IBM objectstore" func=ReconcileDefaultBackingStore sys=openshift-storage-test/noobaa
time="2021-02-09T11:30:51Z" level=info msg="Preparing backing store in IBM Cloud" sys=openshift-storage-test/noobaa
time="2021-02-09T11:30:51Z" level=info msg="✅ Exists:  \"ibm-cloud-cos-creds\"\n"
time="2021-02-09T11:30:51Z" level=info msg="Update event detected for noobaa-endpoint (openshift-storage-test), queuing Reconcile"
time="2021-02-09T11:30:51Z" level=info msg="Constructing endpoint for region: \"eu-gb\"" sys=openshift-storage-test/noobaa
```
```
# oc get backingstore -n openshift-storage-test
NAME                           TYPE      PHASE   AGE
noobaa-default-backing-store   ibm-cos   Ready   2m47s
```
```

time="2021-02-09T11:39:40Z" level=info msg="Reconciling Backing Store Credentials" sys=openshift-storage-test/noobaa
time="2021-02-09T11:39:40Z" level=info msg="Running in IBM Cloud" sys=openshift-storage-test/noobaa
time="2021-02-09T11:39:40Z" level=info msg="❌ Not Found: Secret \"ibm-cloud-cos-creds\"\n"
time="2021-02-09T11:39:40Z" level=info msg="\"ibm-cloud-cos-creds\" secret is not present" sys=openshift-storage-test/noobaa
time="2021-02-09T11:39:40Z" level=info msg="SetPhase: \"Connecting\"" sys=openshift-storage-test/noobaa
```
```
# oc get backingstore -n openshift-storage-test
NAME                           TYPE      PHASE      AGE
noobaa-default-backing-store   pv-pool   Creating   76s
```

File: pkg/util/util.go [hunk 1]
```diff
@@ -837,7 +837,7 @@ func IsIBMPlatform() bool {
 	isIBM := strings.HasPrefix(nodesList.Items[0].Spec.ProviderID, "ibm")
 	if isIBM {
 		// Incase of Satellite cluster is deplyed in user provided infrastructure
-		if strings.Contains(nodesList.Items[0].Spec.ProviderID, "sat-ibm") {
+		if strings.Contains(nodesList.Items[0].Spec.ProviderID, "/sat-") {
```
Comment (jeniawhite) [inline pkg/util/util.go:840]: Do you have a reference to the source that shows the `ProviderID` assignment in IBM cases?
Just trying to verify the change.

**Comment by jeniawhite** [comment]
Please squash the commits to a single commit prior to merging.

File: pkg/util/util.go (hunk 1 above)
Comment (ambiknai) [reply pkg/util/util.go:840]: It is IBM internal code. But I can share how the provider ID looks like
```
$ kubectl get nodes 172.33.16.226 -o yaml | grep -i provider
    ibm-cloud.kubernetes.io/iaas-provider: upi
          f:ibm-cloud.kubernetes.io/iaas-provider: {}
        f:providerID: {}
  providerID: ibm://62db5de4494c64ef4b5f94b057bcc557///c0hbolfw0t9eou3re3p0/sat-ip17233162-15b16ea8f23c4d44d09fb82ec08b21c6a9f85ebc
  ```

**Review by jeniawhite — APPROVED** [review]

**Comment by jeniawhite** [comment]
Please squash the commits to a single commit and rebase with the master.
Thank you.

**Comment by ambiknai** [comment]
> Please squash the commits to a single commit and rebase with the master.
> Thank you.

Done

**Comment by ambiknai** [comment]
@jeniawhite Please could you merge this PR

**Comment by nimrod-becker** [comment]
There are still 2 commits, can you squash to a single commit ?

**Comment by ambiknai** [comment]
@nimrod-becker Please could you merge now

**Pull request merged**